pylint/checkers/refactoring/refactoring_checker.py

```diff
@@ -160,6 +160,25 @@ def _will_be_released_automatically(node: nodes.Call) -> bool:
     return func.qname() in callables_taking_care_of_exit
 
 
+def _is_part_of_assignment_target(node: nodes.NodeNG) -> bool:
+    """Check whether use of a variable is happening as part of the left-hand
+    side of an assignment.
+
+    This requires recursive checking, because destructuring assignment can have
+    arbitrarily nested tuples and lists to unpack.
+    """
+    if isinstance(node.parent, nodes.Assign):
+        return node in node.parent.targets
+
+    if isinstance(node.parent, nodes.AugAssign):
+        return node == node.parent.target
+
+    if isinstance(node.parent, (nodes.Tuple, nodes.List)):
```

Comment:
This will let you handle a set.
```suggestion
    if isinstance(node.parent, nodes.BaseContainer):
```

Reply:
Never mind! You can't unpack into a set. I didn't look closely at my example. So this is purely up to you whether to change this.